src/services/userService.ts

```diff
@@ -0,0 +1,52 @@
+import { User, IUser } from "../models/User";
+import * as mongoose from "mongoose";
+
+export class userService {
+  public static addUser = async (data: IUser): Promise<IUser> => {
+    const newUser = new User({
+      registrationDate: new Date(),
+      ...data,
+    });
+    return new Promise(async (resolve, reject) => {
+      newUser
+        .save()
+        .then((res) => resolve(res))
+        .catch((err) => reject(err));
+    });
+  };
+
+  public static readUser = async (
+    userID: mongoose.Types.ObjectId
+  ): Promise<Array<IUser>> => {
+    return new Promise(async (resolve, reject) => {
+      User.find({ _id: userID })
+        .then((res) => resolve(res))
+        .catch((err) => reject(err));
+    });
+  };
+
+  public static updateUser = async (
+    data: IUser,
+    userID: mongoose.Types.ObjectId
+  ): Promise<IUser> => {
+    return new Promise(async (resolve, reject) => {
+      User.findOneAndUpdate({ _id: userID }, { $set: data }, { new: true })
```

Comment:
why did you use `{ $set: data }` insted of only `data` - just curious

Reply:
Hmm.. I think that I saw it once and it got stuck inside my head. For me it is easier to see what's going on